### ISSUE TITLE
Cannot use Authentication provider Windows with MembershipService Internal

### ISSUE THREAD

**Issue by RedX2501** [jakubgarfield/Bonobo-Git-Server#396]
With a fresh install (empty db) the users retrieved from the authentication provider do not seem to be inserted into the db. So when the Membership provider tries to fetch them it fails.

You also cannot log in as the default admin account as it is not a valid account in the domain.

The part that seems to be failing is in 

https://github.com/jakubgarfield/Bonobo-Git-Server/blob/8faa960af319bc89a5dfc8e93bbe1f081bec3c74/Bonobo.Git.Server/Owin/WindowsAuthenticationHandler.cs#L41

Maybe upon successfully authenticating the user it should be inserted into the DB if not yet present. This would also fix #394 allowing a simple SSO functionality. Maybe password change should be disabled when using Window as an authentication provider.

Pinging @Ollienator as I think it was him who introduced this code.

**Comment by RedX2501** [comment]
I'm currently working in domain ForeignDomain (FD) but my home domain is HomeDomain (HD).

After some more investigating I found out following inconsistent behaviours.

For testing purposes I am editing the DB directly.

Scenario 1:
I have created a user with username `HD\my_user` and same password hash as admin. **Login is accepted**

But I cannot edit my profile as it does not like the slash in the username.

Scenario 2:
Creating a user with username `my_user` and same password as admin. Login is **not** accepted.
There is no way of me passing an username without domain to the login module. IE seems to add it directly. At least it is there when it reprompts me.

I have tried adding a user from the WindowsAuthenticationHandler but I was not successful. Also does it seem the right place to do so.

Some problems I identified:
There is no way of making an initial user admin when using Windows Auth with Internal DB (aside from editing the DB). Maybe we need to reintroduce the "ImportUserAsAdmin" flag from previous Bonobo versions. Or better maybe better "ImportFirstUserAsAdmin".
